### DISCUSSION TITLE
Change default autocompletion key bindings?

### DISCUSSION THREAD

**Discussion by briandipalma** [VonHeikemen/lsp-zero.nvim#121] [Ideas]
Firstly, thanks for the brilliant plugin.

I know this would be a breaking change but I find the current defaults odd in one regard:

```
<Ctrl-u>: Scroll up in the item's documentation.

<Ctrl-f>: Scroll down in the item's documentation.

<Ctrl-d>: Go to the next placeholder in the snippet.

<Ctrl-b>: Go to the previous placeholder in the snippet.
```

To me the mnemonic would suggest "**U**p"/"**D**own" and "**B**ack"/"**F**orward".

Documentation is always going to be moving the scrollbar up/down, while placeholders could be on the same line so going back and forward between placeholders makes sense for them.

Given that would these shortcuts not be more intuitive?

```
<Ctrl-u>: Scroll up in the item's documentation.

<Ctrl-d>: Scroll down in the item's documentation.

<Ctrl-f>: Go forward to the next placeholder in the snippet.

<Ctrl-b>: Go back to the previous placeholder in the snippet.
``` 

I know I can override these, and I will locally, but I feel these two shortcuts should be swapped around in the core plugin due to the mnemonics/user intuition.

**Comment by VonHeikemen** [comment]
I had reason for this but I can't remember. Maybe I just thought `Ctrl+d` was more comfortable to type. I don't scroll the documentation window very often, maybe keeping `Ctrl+d` for it felt like a waste.

If I ever do a breaking change, I'll consider this.

**Comment by briandipalma** [comment]
I think it's because the documentation for `nvim-cmp` uses those key bindings: https://github.com/hrsh7th/nvim-cmp/ the `nvim-cmp` binding examples don't have snippet placeholder navigation and the same quality/style documentation you have which I think made the underlying awkwardness of those two shortcuts more obvious.

**Reply by VonHeikemen** [reply]
Oh I remember now, I configured nvim-cmp using the [wiki in nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion#nvim-cmp). They have `Ctrl+d` and `Ctrl+f` to scroll the docs. I thought using `Ctrl + d` to go up was odd then change it to `Ctrl + u`. So its `f` to go forward and `u` to go up. For the snippet navigation just used whatever shortcut was available (in my personal config) that it was easy to type.

**Comment by briandipalma** [comment]
If you look at the man pages for Neovim Ctrl-u/d/b/f are using the mnemonics (up/down/back/forward) I mentioned, I'm guessing that's where I got the idea.
